commit db95dd53f88bb15e288b553de5c6687260756f03
Author: Willy Tarreau <w@1wt.eu>
Date:   Tue Feb 12 10:59:32 2019 +0100

    BUILD/MINOR: stream: avoid a build warning with threads disabled
    
    gcc 6+ complains about a possible null-deref here due to the test in
    objt_server() :
    
           if (objt_server(s->target))
               HA_ATOMIC_ADD(&objt_server(s->target)->counters.retries, 1);
    
    Let's simply change it to __objt_server(). This can be backported to
    1.9 and 1.8.
    
    (cherry picked from commit 1ef724e2169eaff7f0272278c3fba9b34d5c7f78)
    Signed-off-by: Willy Tarreau <w@1wt.eu>
    (cherry picked from commit 3d3b67f1877718abbbc8cc500aae373640e454e9)
    Signed-off-by: Willy Tarreau <w@1wt.eu>

diff --git a/src/stream.c b/src/stream.c
index f443cc74..99a133a9 100644
--- a/src/stream.c
+++ b/src/stream.c
@@ -746,7 +746,7 @@ static int sess_update_st_cer(struct stream *s)
 		si->state = SI_ST_REQ;
 	} else {
 		if (objt_server(s->target))
-			HA_ATOMIC_ADD(&objt_server(s->target)->counters.retries, 1);
+			HA_ATOMIC_ADD(&__objt_server(s->target)->counters.retries, 1);
 		HA_ATOMIC_ADD(&s->be->be_counters.retries, 1);
 		si->state = SI_ST_ASS;
 	}
